### PR TITLE
Relocate inline JS code in graph plugin to separate file

### PR DESCRIPTION
**Pull request by syncguru**
Fixes #21669

The approach adapted for this fix is to use html data attributes  .. for more details, see https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes

### CODE REVIEW

File: plugins/MantisGraph/MantisGraph.php [hunk 1]
```diff
@@ -113,6 +113,7 @@ function resources() {
 			html_javascript_link( 'chart-' . CHARTJS_VERSION . '.min.js' );
 			html_javascript_link( 'chart.bundle-' . CHARTJS_VERSION . '.min.js' );
 		}
+		html_javascript_link( '../plugins/MantisGraph/MantisGraph.js' );
```
Comment (vboctor) [inline plugins/MantisGraph/MantisGraph.php]: Move the MantisGraph.js file into plugins/MantisGraph/files and include it by getting its location using plugin_file( 'MantisGraph.js' ).  See Snippets plugin for an example.

**Comment by vboctor** [comment]
There seems to be some javascript errors on the login page:

javascript_error_on_login_page

File: plugins/MantisGraph/MantisGraph.php (hunk 1 above)
Comment (cproensa) [inline plugins/MantisGraph/MantisGraph.php]: Should chart js libs, ("chart-xxxx.min.js", etc) also be provided by the plugin?

**Comment by cproensa** [comment]
> There seems to be some javascript errors on the login page

i noticed those when doing the work for verify page.. (but forgot to follow on them)
I don't think they are directly related to this PR

File: plugins/MantisGraph/MantisGraph.php (hunk 1 above)
Comment (vboctor) [inline plugins/MantisGraph/MantisGraph.php]: Currently chart-xxxx.min.js files are in the js folder for core.  They should be moved into plugins/MantisGraph/files as well.

Comment (cproensa) [inline plugins/MantisGraph/MantisGraph.php]: > We could use chart-xxxx.min.js for some dashboard functionality (issue trend, ...) on "My View" page. So maybe better not to move.

That was my question about. I don't know if they are used system-wide.
In that case, the inclusion of those js, should be done by core, not by the plugin?

Comment (dregad) [inline plugins/MantisGraph/MantisGraph.php]: Unless they are actually used somewhere else, I would rather move all js files to the plugin's file directory.

We can revisit this if we realize later on that some of these scripts are used elsewhere.

One could also argue that a dashboard should be a part of the Graph plugin itself.

**Comment by dregad** [comment]
>There seems to be some javascript errors on the login page

This is addressed in PR #922 (see [#21805](https://www.mantisbt.org/bugs/view.php?id=21805))

**Comment by syncguru** [comment]
@vboctor, @cproensa, @dregad 
+ Moved JS into plugin files directory
+ Removed CSP cdoe

**Pull request merged**